### PR TITLE
Refactoring menus

### PR DESCRIPTION
**Pull request by krainboltgreene**
Switching to a more compact and local menu system

### CODE REVIEW

File: browser/lowbrow/menu.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
+const modifier = if (process.platform == "darwin") {
```
Comment (travisbhartwell) [inline browser/lowbrow/menu.js:1]: Awesome you can assign the result of an if statement in Javascript.  Big feature from Haskell I miss in other languages, notably Python.

File: browser/lowbrow/menu.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  return "Command";
+} else {
+  return "Control";
+}
+
+const seperator = {
```
Comment (travisbhartwell) [inline browser/lowbrow/menu.js:7]: Nit-pick.  `seperator` is spelled `separator` :)

File: browser/lowbrow/menu.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  type: "seperator"
+}
+
+var shortcut = function() {
+  retrun Array.prototype.slice.call(arguments).join("+")
```
Comment (travisbhartwell) [inline browser/lowbrow/menu.js:12]: s/`retrun`/`return`/

File: browser/lowbrow/menu.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    label: label,
+    accelerator: accelerator,
+    selector: selector,
+    submenu: submenu,
+    click: click
+  };
+}
+
+var menuGroup = function(label, items) {
+  return menuItem(label, null, null, items)
```
Comment (travisbhartwell) [inline browser/lowbrow/menu.js:26]: I hate uses of null as parameters.  If I'm not a heavy user of an API, it's not clear why you are passing null.  To be more clear, how about creating constants like these:

``` Javascript
const NO_ACCELERATOR = null;
const NO_SELECTOR = null;
```

And then change this line, and lines like it below, to:

``` Javascript
  return menuItem(label, NO_ACCELERATOR, NO_SELECTOR, items);
```

Seems clearer to me.

**Comment by travisbhartwell** [comment]
I made a few inline comments, but otherwise enjoy the abstraction.  Looks clean.

File: browser/lowbrow/menu.js (hunk 4 above)
Comment (krainboltgreene) [inline browser/lowbrow/menu.js:26]: Love this idea. Want to make this PR? Otherwise I'll hit it up tonight.

Comment (travisbhartwell) [inline browser/lowbrow/menu.js:26]: Yeah, I could do so, I'll do it this afternoon.

**Pull request merged**